### ISSUE TITLE
Conv2d weight matrix size is reversed?

### ISSUE THREAD

**Issue by abhih1** [keras-team/keras#8144]
Hi,
I'm trying to get the weight matrix of a conv2d layer. This is the snippet of the code:

`model = Sequential()

model.add(Conv2D(filters=16, kernel_size=(3,3),
                        input_shape=(9,9,1),activation='relu',kernel_regularizer = l2(regularization_coef)))

model.add(Conv2D(filters=64, kernel_size=(3,3),activation='relu',kernel_regularizer = l2(regularization_coef)))

model.add(MaxPooling2D(pool_size=(2,2)))
model.add(Dropout(0.5))

model.add(Flatten())
model.add(Dense(128,activation='relu',kernel_regularizer = l2(regularization_coef)))

model.add(Dropout(0.5))
model.add(Dense(2,activation='softmax',kernel_regularizer = l2(regularization_coef)))

model.compile(loss='categorical_crossentropy', optimizer='adadelta',metrics=['accuracy'])
model.summary()
#plot_model(model, to_file='model.png',show_shapes=True,show_layer_names=True)
model.fit(X_train, Y_train, batch_size=batch_size, epochs=nb_epoch, verbose=0, validation_split=0.1)

score = model.evaluate(X_test, Y_test, verbose=0)
print('Test score:', score[0])
print('Test accuracy:', score[1])

filters= model.layers[0].get_weights()[0][:,:,0,:]
print(filters)
print(filters.shape)`

Now, my layers[0] has 16 filters, of kernel size 3x3 each, and the shape of the filters should be (16,3,3). But I'm getting an array of shape (3,3,16). Can someone please help me with this? I'm trying to visualise the weight matrix, and am running into problems because of this reversed shapes.

**Comment by kgrm** [comment]
That's the tensorflow convention, the kernel shape is `(kernel_height, kernel_width, input_channels, output_channels)`. To reverse it, you can always just get the value and transpose it appropriately, e.g. with `filters = model.get_weights()[0][:, :, 0, :].transpose((2, 1, 0))`.

**Comment by ymodak** [comment]
Closing as this is resolved. Feel free to reopen if have any follow up questions.
